.github/scripts/tag-stage.sh

```diff
@@ -0,0 +1,193 @@
+  # where "VARIANT" is of the form: <distro>-<codename> OR <library>-<library version>  (ex: "ubuntu-jammy", "gnome-43")
+  # complete examples: "r4-ubuntu-jammy", "r3_1-beta2-debian-bullseye"
+  if [ "$PACKAGE_SOURCE_REF" == "main" ]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [ "$PACKAGE_SOURCE_REF" == "master" ]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [ "$PACKAGE_SOURCE_REF" == "debian-bullseye" ]; then
+    tag_package "$DEFAULT_DEST_TAG-debian-bullseye"
+  elif [[ "$PACKAGE_SOURCE_REF" == "debian-testing" || "$PACKAGE_SOURCE_REF" == "debian/testing" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-debian-testing"
+  elif [[ "$PACKAGE_SOURCE_REF" == "debian-bookworm" || "$PACKAGE_SOURCE_REF" == "debian-bookworm-compat" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-debian-bookworm"
+  elif [ "$PACKAGE_SOURCE_REF" == "regolith/1%43.0-1" ]; then
+    tag_package "$DEFAULT_DEST_TAG-gnome-43"
+  elif [ "$PACKAGE_SOURCE_REF" == "regolith/46" ]; then
+    tag_package "$DEFAULT_DEST_TAG-gnome-46"
+  elif [[ "$PACKAGE_SOURCE_REF" == "ubuntu-jammy" || "$PACKAGE_SOURCE_REF" == "ubuntu/jammy" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-ubuntu-jammy"
+  elif [[ "$PACKAGE_SOURCE_REF" == "ubuntu-focal" || "$PACKAGE_SOURCE_REF" == "ubuntu/focal" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-ubuntu-focal"
+  elif [[ "$PACKAGE_SOURCE_REF" == "debian-v9" && "$PACKAGE_NAME" == "picom" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [[ "$PACKAGE_SOURCE_REF" == "debian" && "$PACKAGE_NAME" == "whitesur-gtk-theme" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [[ "$PACKAGE_SOURCE_REF" == "applied/ubuntu/groovy" && "$PACKAGE_NAME" == "xcb-util" ]]; then
+    : # this package is exceptional, is not built from a regoolith repo.  cannot push tags.  deprecated.
+  elif [[ "$PACKAGE_SOURCE_REF" == "ubuntu/v0.32.1" && "$PACKAGE_NAME" == "i3status-rs" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [[ "$PACKAGE_SOURCE_REF" == "ubuntu/v0.22.0" && "$PACKAGE_NAME" == "i3status-rs" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-ubuntu-jammy"
+  elif [[ "$PACKAGE_SOURCE_REF" == "i3cp" && "$PACKAGE_NAME" == "regolith-rofi-config" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [[ "$PACKAGE_SOURCE_REF" == "packaging/v1.7-regolith" && "$PACKAGE_NAME" == "sway-regolith" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-ubuntu-jammy"
+  elif [[ "$PACKAGE_SOURCE_REF" == "packaging/v1.8-regolith" && "$PACKAGE_NAME" == "sway-regolith" ]]; then
+    tag_package "$DEFAULT_DEST_TAG-debian-testing"
+  elif [[ "$PACKAGE_SOURCE_REF" == "packaging/v1.9-regolith" && "$PACKAGE_NAME" == "sway-regolith" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  elif [[ "$PACKAGE_SOURCE_REF" == "debian" && "$PACKAGE_NAME" == "fonts-nerd-fonts" ]]; then
+    tag_package "$DEFAULT_DEST_TAG"
+  else
+    echo "### Ignoring unhandled variant: $PACKAGE_NAME on $STAGE-$DISTRO-$CODENAME with tag $PACKAGE_SOURCE_REF"
+  fi
+
+  popd > /dev/null
+}
```

Comment:
This seems like a hassle. Maybe we can eventually migrate to a consistent naming strategy for the entire project?

Reply:
> ...consistent naming strategy for the entire project?

Yes, in fact this PR is the first iteration to achieve that goal.  It's been a tedious and error prone process to do each release, as I've had to go into the package model and fix things up from memory.  This script will allow the first part of automating this: cut tags for a release based upon package model.  For 3.2 I'll stick to this.  For the next release I'll look into automating mutating the next-stage package model w/ the data.  Once that's working then releases should be automated and consistent.

Reply:
This sounds reasonable.

Reply:
I've done some more work to document conventions in the script and further categories branch mappings as either conventions or special cases.